src/custom/state/claim/hooks/index.ts

```diff
@@ -294,11 +305,19 @@ function useDeploymentTimestamp(): number | null {
       return
     }
 
-    vCowContract.deploymentTimestamp().then((ts) => {
-      console.log(`Deployment timestamp in seconds: ${ts.toString()}`)
-      setTimestamp(ts.mul('1000').toNumber())
-    })
-  }, [chainId, vCowContract])
+    if (timestamp) {
+      return
+    }
+
+    fetchDeploymentTimestamp(vCowContract)
+      .then((timestamp) => {
+        setTimestamp(timestamp)
```

Comment:
What happens if the hook is destroyed? wouldn't this set the timestamp and create a memory leak?

Reply:
Updated

Reply:
I didn't see any change. Doesn't this generate the same memory leaks you were trying to solve?